### PR TITLE
move api key to configuration file

### LINKED ISSUE #99: YouTube API key as config option
I had some issues getting the search feature to work, and finally solved it by changing the youtube API key directly in the backend.py script, but i think it would be easier and also would help others facing the same problem if you coul change the API key from the config files .
Something like this:

`[youtube]`
`api_key = '<key_goes_here>'`

### PR DESCRIPTION
**Pull request by natumbri**
closes #99

### CODE REVIEW

**Comment by brammittendorff-dd** [comment]
@natumbri Nice to see these fixes, maby you should add youtube_dl to the travic-ci:
```
install: - "pip install tox && pip install youtube_dl"
```

**Comment by natumbri** [comment]
Thanks for your feedback. I'll have a look and see if I can fix the things you've mentioned.

I didn't actually mean to submit all of those changes at once, in a single go. I'm not even sure how it happened - I'll try to work that out too!

I should say, I'm not that keen to spend time trying to come up with patch requests against the existing code in circumstances that they will not ultimately be useful (because it will be superseded by more fundamental changes are about to be made). I started working from the fork @chatziko created, which I found to offer better performance, and to be written in a way that I could more easily work with. All the changes I have made are against that fork. Is there any way to get the more wholesale changes accepted as a kind of 'reset'?

File: mopidy_youtube/backend.py [hunk 1]
```diff
@@ -139,7 +138,8 @@ def resolve_playlist(url):
 class YouTubeBackend(pykka.ThreadingActor, backend.Backend):
     def __init__(self, config, audio):
         super(YouTubeBackend, self).__init__()
-        self.config = config
+        ytconf = config['youtube']
+        yt_key = ytconf['api_key']
```
[the remark this answers is not in the document]
Comment (natumbri) [reply mopidy_youtube/backend.py]: Thanks for taking the time to look at this. You are quite right - it doesn't work! Sorry! I was trying to work backwards from the version I am actually using (which works) to reverse engineer a patch to the current official version. What you've suggested sounds like a good plan.

**Comment by natumbri** [comment]
OK, I think it will be easier if I start this from scratch, so I'm going to close and start again.  I'll take on board all the helpful comments from @kingosticks and @brammittendorff-dd before I put something in again.

**Pull request closed without merging**